pgx/src/datum/array.rs

```diff
@@ -414,11 +389,16 @@ where
     fn type_oid() -> u32 {
         unsafe { pg_sys::get_array_type(T::type_oid()) }
     }
+
+    #[inline]
+    fn is_compatible_with(other: pg_sys::Oid) -> bool {
+        Self::type_oid() == other || other == unsafe { pg_sys::get_array_type(T::type_oid()) }
+    }
 }
 
 impl<'a, T> IntoDatum for &'a [T]
 where
-    T: IntoDatum + Copy,
+    T: IntoDatum + Copy + 'a,
```

Comment:
I wonder why this now needs a lifetime annotation here?

Reply:
```
error[E0309]: the parameter type `T` may not live long enough
   --> pgx/src/datum/array.rs:442:9
    |
399 | impl<'a, T> IntoDatum for &'a [T]
    |          - help: consider adding an explicit lifetime bound...: `T: 'a`
...
442 |         Self::type_oid() == other || other == unsafe { pg_sys::get_array_type(T::type_oid()) }
    |         ^^^^ ...so that the reference type `&'a [T]` does not outlive the data it points at
```